nfs/src/nfs_client.cpp

```diff
@@ -1554,13 +1554,32 @@ void nfs_client::reply_entry(
         AZLogDebug("[{}] <{}> Returning ino {} to fuse (filename: {}, "
                    "lookupcnt: {}, dircachecnt: {}, forget_expected: {})",
                    parent_ino,
-                   rpc_task::fuse_opcode_to_string(ctx->rpc_api->optype),
+                   rpc_task::fuse_opcode_to_string(task->rpc_api->optype),
                    inode->get_fuse_ino(),
-                   ctx->rpc_api->get_file_name(),
+                   task->rpc_api->get_file_name(),
                    inode->lookupcnt.load(),
                    inode->dircachecnt.load(),
                    inode->forget_expected.load() + 1);
 
+        if (optype == FUSE_LOOKUP) {
+            /*
+             * LOOKUP RPC can be issued in one of the following 2 cases:
+             * case1: The fuse has made a lookup call to get the filehandle.
+             *        This is the most common case.
+             * case2: Fuse issued create/mknod/mkdir calls returned
+             *        successfully but the NFS server failed to populate the
+             *        filehandle/postop attributes. In such cases, we make a
+             *        lookup call to get this filehandle as the fuse expects
+             *        it in response to the create/mknod/mkdir calls.
+             *        In this case, we need to return appropriate reply to
+             *        the fuse depending on what operation caused this lookup
+             *        call to be triggered. This information is stored in the
+             *        'called_for_optype' member of the lookup task and will
+             *        be set to that opcode of the calling op.
+             */
+            optype = task->get_proxy_op_type();
```

Comment:
this must be done before the above log so that we correctly report the original task on behalf of which we are returning the inode.